### PR TITLE
#398 - Implement styles for tabbed navigation

### PR DESCRIPTION
**Pull request by blms**
## Visual review notes
[Percy build](https://percy.io/2cf28a24/geniza/builds/14882339)

In scope:
- Tabbed navigation in document detail/scholarship views

## What this PR does
- Per #398:
  - Implements styles for default, selected, hover/active, and disabled tabs in the document detail/scholarship records tabbed navigation

### CODE REVIEW

**Comment by blms** [comment]
@rlskoeser Requesting your review because I wanted to ask: how do you feel about the use of elements (rather than pseudoelements) for the tab navigation underlines? Not semantically useful, but the best I could come up with to get it working consistently.

I really wanted to use `::after` pseudoelements for them, but the problem is that when relatively-positioned, their position is dependent on the preceding content—rather than the dimensions of the container. Since different browsers seem to render the text content at different heights, I couldn't get the individual hover/active underlines to align consistently with the gray persistent underline. 

This would be fine if we just positioned them absolutely, but in that case, their widths must be explicitly defined. And since the tabs have different widths, they need to inherit width from the tab `li`s! I also considered using `::after` on the `li` instead of the `a` and `span`, but then you can't depend on hover/active states to switch styles.

**Comment by blms** [comment]
@gissoo This one is ready for visual review, notes posted in PR description above!

**Comment by gissoo** [comment]
@blms there is no way for me to approve the build – what would you like me to do about the percy pages?

**Comment by gissoo** [comment]
@blms I tested the tabs on the test site – the click works well however the tab style still has not been implemented, correct? The bar is missing and it looks like there is a link style on these, because they both have underlines instead of the one that's active (like in the percy pages) – otherwise it works well!

**Comment by blms** [comment]
> @blms I tested the tabs on the test site – the click works well however the tab style still has not been implemented, correct? The bar is missing and it looks like there is a link style on these, because they both have underlines instead of the one that's active (like in the percy pages) – otherwise it works well!

@gissoo That's right, it's not ready in QA yet. Sorry for being unclear there! For now, please just review what you can see on Percy (see Slack for messages about access). I'll tag you in the issue #398 once it's ready on the QA site to test interactions.

**Comment by blms** [comment]
@rlskoeser I hadn't thought of just using `border-bottom` on the `a` tags! That works, except it doesn't allow for the "extra bit" of gray outline on either side of the tabs (i.e. the width of the `border-bottom` in the designs is more like 80% of the width of the tab).

And the focus style is still a bit weird—let me know if you have any ideas for what to do about that.

Here's what things look like (hover, then focus, shown on left tab) using the `border-bottom` solution:

Screen Shot 2022-01-10 at 12 07 29 PM
Screen Shot 2022-01-10 at 12 07 36 PM
 
Also set the tabs to be equally sized. You're right, there's no reason they shouldn't be!

@gissoo Are you OK with this appearance? There's no "extra bit" on either side of the tabs, and the focus style doesn't exactly match the designs.

**Comment by rlskoeser** [comment]
This looks good to me @blms ! And the focus style seems ok.

I like the idea of making the links & underlines full width within the tabs — it seemed like when I was testing it before there was a dead zone between tabs that wasn't clickable, I imagine that wouldn't be the case any more? 

Interested to hear what @gissoo thinks.

**Comment by blms** [comment]
@rlskoeser Right, there'd be no "dead zone" with this design!

**Comment by gissoo** [comment]
@rlskoeser @blms I'm ok with the tab decision, that is fine! I'm not really ok with the focus styles but it's not a deal breaker, wouldn't prioritize spending time on it right now.

File: sitemedia/scss/components/_tabs.scss [hunk 1]
```diff
@@ -2,35 +2,58 @@
...
-    }
-    // Center tab padding boost
-    li:nth-child(2) a,
-    li:nth-child(2) span {
-        @include breakpoints.for-tablet-landscape-up {
-            padding: spacing.$spacing-2xs spacing.$spacing-2xl;
+        text-decoration: none;
+        &:focus-visible {
+            outline: 0.1rem solid var(--tabs-focus);
+            outline-offset: -0.33rem;
```
Comment (rlskoeser) [inline sitemedia/scss/components/_tabs.scss:35]: @blms I had an idea about using a pseudo-element to get the focus outline closer to the designs. I tried this out locally, see what you think.


```suggestion
      position: relative;
       /* use ::after for focus outline */
        &::after {
            content: "";
            width: 80%;
            height: 80%;
            position: absolute;
        }
        &:focus-visible {
            outline: none;
        }
        &:focus-visible::after {
            outline: 0.1rem solid var(--tabs-focus);
        }
```

Comment (blms) [reply sitemedia/scss/components/_tabs.scss:35]: This is great!! Excellent solution, Rebecca.

**Comment by blms** [comment]
Thanks @gissoo and @rlskoeser! Went ahead and approved the Percy build after Rebecca's latest changes, since Gissoo already approved the prior build. Merging now.

**Pull request merged**